frontend/src/components/pages/RegistrantExperience/index.tsx

```diff
@@ -109,62 +166,169 @@ const RegistrantExperiencePage = (): React.ReactElement => {
           />
         );
       case RegistrantExperienceSteps.ReviewRegistrationPage:
-        return (
-          <ReviewRegistration
-            isChecked={sampleRegisterField}
-            toggleChecked={() => setSampleRegisterField(!sampleRegisterField)}
-            campName={camp?.name || ""}
-          />
-        );
+        if (camp) {
+          return (
+            <ReviewRegistration campers={registeringCampers} camp={camp} />
+          );
+        }
+        return <></>;
       default:
         throw new Error("unexpected page");
     }
   };
 
+  const isCurrentStepCompleted = (step: RegistrantExperienceSteps) => {
+    switch (step) {
+      case RegistrantExperienceSteps.PersonalInfoPage:
+        return isPersonalInfoFilled;
+      case RegistrantExperienceSteps.AdditionalInfoPage:
+        return isAdditionalInfoFilled;
+      case RegistrantExperienceSteps.WaiverPage:
+        return isWaiverFilled;
+      case RegistrantExperienceSteps.ReviewRegistrationPage:
+        return isReviewRegistrationFilled;
+      default:
+        return false;
+    }
+  };
+
   const handleStepNavigation = (stepsToMove: number) => {
     const desiredStep = currentStep + stepsToMove;
     if (RegistrantExperienceSteps[desiredStep]) {
       setCurrentStep(currentStep + stepsToMove);
     } else if (desiredStep < 0) {
       alert("PLACEHOLDER - go to initial registration");
+    } else if (camp) {
+      registerCampers(registeringCampers, camp);
     } else {
-      alert("PLACEHOLDER - go to payment");
+      alert("Unexpected camp state. Please restart registration.");
```

Comment:
Not sure how we would reach this state. If it's possible, I think we should rather avoid it than show this message as it's kinda confusing how the error even happened